### PR TITLE
Bug 1811695: Add missing optional tags

### PR DESCRIPTION
**Pull request by soltysh**
This is a cherry-pick of https://github.com/openshift/api/pull/511 onto 3.11 release.

/assign @sttts

### CODE REVIEW

**Comment by sttts** [comment]
@soltysh how clean was the cherry-pick?

**Comment by soltysh** [comment]
> @soltysh how clean was the cherry-pick?

The pick was clean but I can't run go1.10 locally :sweat: and that is required to get the formatting digestible by our CI.

File: template/v1/types.go [hunk 1]
```diff
@@ -127,6 +128,7 @@ type TemplateInstanceSpec struct {
 
 	// requester holds the identity of the agent requesting the template
 	// instantiation.
+	// +optional
```
Comment (sttts) [inline template/v1/types.go]: what about secret above?

**Pull request merged**